web/src/pages/api/public/ingestion.ts

```diff
@@ -212,6 +212,36 @@ export default async function handler(
 
     const sortedBatch = sortBatch(batch);
 
+    // We group events by eventBodyId which allows us to store and process them
+    // as one which reduces infra interactions per event. Only used in the S3 case.
+    const sortedBatchByEventBodyId = sortedBatch.reduce(
```

Comment:
@maxdeichmann This will store all events by body id in order, but we won't have a guaranteed global order. Is that ok? My understanding is that only events with the same eventBodyId need to be processed in the correct order.

Reply:
Correct, only events with the same eventBodyId need to e processed in the same order.